src/types/helpers.tspec.ts

```diff
@@ -1,8 +1,34 @@
 import { AssertTrue, IsExact } from 'conditional-type-checks';
-import { ActionCreatorWithPayload } from './ActionCreator';
-import { ExtractPayload } from './helpers';
+import { ActionCreator, ActionCreatorWithPayload } from './ActionCreator';
+import { ExtractPayload, InferPayloadFromActionCreator } from './helpers';
 
 type Payload = { foo: number };
+type VoidFn = () => void;
+
 type ExtractPayload_extracts_payload = AssertTrue<
   IsExact<ExtractPayload<ActionCreatorWithPayload<Payload>>, Payload>
 >;
+type ExtractPayload_extracts_payload2 = AssertTrue<
+  IsExact<ExtractPayload<ActionCreatorWithPayload>, void>
+>;
+// type aaa = ExtractPayload<VoidFn>; // weakness of ExtractPayload
+// type aaa3 = ExtractPayload<unknown>; // weakness of ExtractPayload
```

Comment:
InferPayloadFromActionCreator is the same as ExtractPayload, although a bit more flexible.
Thinking of renaming the former to the latter, so we use the most flexible version.